bluld: add new package

ktgeek opened the pull request:
Signed-off-by: Keith T. Garner <kgarner@kgarner.com>

Maintainer: me
Compile tested: x86-64, 19.07.7, 21.02-rc1 
Run tested: x86-64, 19.07.7, tested by running for a year+, tested by running without a config, memory profiling

Description: A small daemon using the user-space LED driver to expose Blinkstick LEDs in /sys/class/leds

PolynomialDivision commented:
If you want to merge the package into master, you also should compile and runtest on master. ;)

PolynomialDivision commented on utils/bluld/Makefile:
```diff
@@ -0,0 +1,47 @@
+endef
+
+$(eval $(call BuildPackage,bluld))
+
```
Why this extra newline?

ktgeek replied:
I'll fix that up.

PolynomialDivision commented on utils/bluld/Makefile:
```diff
@@ -0,0 +1,47 @@
+endef
+
+define Package/bluld/description
+A daemon to expose a Blinkstick to the Linux kernel LED class via /dev/uleds. (Kernel 4.10+ required.)
```
This should be intended.

PolynomialDivision replied:
Sorry. I mean indent. indentation xD

ktgeek commented:
> If you want to merge the package into master, you also should compile and runtest on master. ;)

Will do!

PolynomialDivision commented on utils/bluld/Makefile:
```diff
@@ -0,0 +1,47 @@
+PKG_HASH:=2622d5dcddf2712c934d718a67116cbf4fba5327ae2873421e0df518c7d80fe1
+
+PKG_MAINTAINER:=Keith T. Garner <kgarner@kgarner.com>
+PKG_LICENSE:=MIT
```
You could also add: ```PKG_LICENSE_FILES:=LICENSE```

PolynomialDivision commented on utils/bluld/files/bluld.init:
```diff
@@ -0,0 +1,35 @@
+}
+
+reload_service() {
+  stop
+  start
+  (sleep 1 && /etc/init.d/led restart) >/dev/null 2>&1 &
+}
```
The number of spaces is different to all others functions.

PolynomialDivision commented on utils/bluld/files/bluld.init:
```diff
@@ -0,0 +1,35 @@
+}
+
+reload_service() {
+  stop
+  start
+  (sleep 1 && /etc/init.d/led restart) >/dev/null 2>&1 &
```
For me this line looks a bit strange.

PolynomialDivision commented on utils/bluld/files/bluld.init:
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2020 Keith T. Garner
```
We have 2021. ;)

ktgeek replied:
but I haven't touched the file in a year :) lol

ktgeek commented:
Thanks for all the feedback.  I'm going to look into the compile error on arc, confirm test on master, and then push a an update.  (It'll probably take a few days.)

PolynomialDivision commented on utils/bluld/files/bluld.init:
```diff
@@ -0,0 +1,39 @@
+    procd_add_reload_trigger "bluld"
+}
+
+reload_service() {
+    stop
+    start
+    /etc/init.d/led restart
+}
+
+restart() {
+    reload_service
+}
```
Never saw something like this? (just asking)

ktgeek replied:
I saw that pattern in another package (and that was two weeks ago, so I forget which) and that seemed to do what I needed.  In this particular case, reload and restart are the same thing: nuke the process and start over (and restart the LED settings on the assumption we're changing an LED color.)

Or am I misunderstanding your question

PolynomialDivision replied:
Okay, than I'm fine.

PolynomialDivision replied:
Could be xray-core. ;)

ktgeek commented:
Ok, things should be in a good place now.  I fixed the arc_archs compile issue, and I think I addressed all the previous comments.  (And I see one was just left on the new init... off to answer.)

PolynomialDivision commented:
lgtm. thanks for your contribution! :)

ktgeek commented:
> lgtm. thanks for your contribution! :)

Thanks for the great feedback on the first pass!
